File: Calc_Android/java/com/reflectcode/demo/project/calculator/MainActivity.java
package com.reflectcode.demo.project.calculator;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.app.Activity;
import android.text.Editable;
import android.util.Log;
import android.view.Menu;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;


public class MainActivity extends AppCompatActivity implements View.OnClickListener  {

    Button one, two, three, four, five, six, seven, eight, nine, zero, add, sub, mul, div, cancel, equal;
    TextView disp, equation;
    int op1, op2;
    Boolean continueCalculationl, clearDisplay;
    String optr;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        continueCalculationl = false; clearDisplay = false;
        optr = "";
        op1=0; op2=0;

        one = (Button) findViewById(R.id.one);
        two = (Button) findViewById(R.id.two);
        three = (Button) findViewById(R.id.three);
        four = (Button) findViewById(R.id.four);
        five = (Button) findViewById(R.id.five);
        six = (Button) findViewById(R.id.six);
        seven = (Button) findViewById(R.id.seven);
        eight = (Button) findViewById(R.id.eight);
        nine = (Button) findViewById(R.id.nine);
        zero = (Button) findViewById(R.id.zero);
        add = (Button) findViewById(R.id.add);
        sub = (Button) findViewById(R.id.sub);
        mul = (Button) findViewById(R.id.mul);
        div = (Button) findViewById(R.id.div);
        cancel = (Button) findViewById(R.id.cancel);
        equal = (Button) findViewById(R.id.equal);
        disp = (TextView) findViewById(R.id.display);
        equation = (TextView) findViewById(R.id.equation);

        try{
            one.setOnClickListener(this);
            two.setOnClickListener(this);
            three.setOnClickListener(this);
            four.setOnClickListener(this);
            five.setOnClickListener(this);
            six.setOnClickListener(this);
            seven.setOnClickListener(this);
            eight.setOnClickListener(this);
            nine.setOnClickListener(this);
            zero.setOnClickListener(this);
            cancel.setOnClickListener(this);
            add.setOnClickListener(this);
            sub.setOnClickListener(this);
            mul.setOnClickListener(this);
            div.setOnClickListener(this);
            equal.setOnClickListener(this);
        } catch(Exception e){

        }
    }


    public void operation(){
        if(!optr.equals("")) {
            op2 = Integer.parseInt(disp.getText().toString());
            equation.setText(equation.getText() + Integer.toString(op1) + " " + optr + " " + Integer.toString(op2) + " | ");

            if (optr.equals("+")) {
                op1 = op1 + op2;
                disp.setText(Integer.toString(op1));
            } else if (optr.equals("-")) {
                op1 = op1 - op2;
                disp.setText(Integer.toString(op1));
            } else if (optr.equals("*")) {
                op1 = op1 * op2;
                disp.setText(Integer.toString(op1));
            } else if (optr.equals("/")) {
                op1 = op1 / op2;
                disp.setText(Integer.toString(op1));
            }
            op2 = 0;
            continueCalculationl = true;
        }
    }
    public void updateDisp(CharSequence str) {
        //Editable strToUpdate = (Editable)disp.getText();
        //Editable strToUpdate;
        String strToUpdate;

        strToUpdate = (String) disp.getText();

        if (strToUpdate.toString().equals("0")) {
            strToUpdate= "";
        }

        if (clearDisplay == true) {
            clearDisplay = false;
            strToUpdate = "" ;
        }

        strToUpdate = strToUpdate + str;
        disp.setText(strToUpdate);
    }


    @Override public void onClick(View btnPressed) {
        switch(btnPressed.getId()){
            case R.id.zero:
                updateDisp(zero.getText());
                break;

            case R.id.one:
                updateDisp(one.getText());
                break;

            case R.id.two:
                updateDisp(two.getText());
                break;

            case R.id.three:
                updateDisp(three.getText());
                break;

            case R.id.four:
                updateDisp(four.getText());
                break;

            case R.id.five:
                updateDisp(five.getText());
                break;

            case R.id.six:
                updateDisp(six.getText());
                break;

            case R.id.seven:
                updateDisp(seven.getText());
                break;

            case R.id.eight:
                updateDisp(eight.getText());
                break;

            case R.id.nine:
                updateDisp(nine.getText());
                break;

            case R.id.cancel:
                op1 = 0;
                op2 = 0;
                continueCalculationl = false;
                clearDisplay = true;
                disp.setText("0");
                disp.setHint("Perform Operation :)");
                equation.setText("");
                break;

            case R.id.add:
                if(op1 == 0){
                    op1 = Integer.parseInt(disp.getText().toString());
                    disp.setText("0");
                    optr = "+";
                } else {
                    if (continueCalculationl == false){
                        disp.setText("0");
                        optr = "+";
                        continueCalculationl = true;
                    }
                    else {
                        operation();
                        optr = "+";
                        clearDisplay = true;
                    }
                }
                break;

            case R.id.sub:
                if(op1 == 0){
                    op1 = Integer.parseInt(disp.getText().toString());
                    disp.setText("0");
                    optr = "-";
                }  else{
                    if (continueCalculationl == false){
                        disp.setText("0");
                        optr = "-";
                        continueCalculationl = true;
                    }
                    else {
                        operation();
                        optr = "-";
                        clearDisplay = true;
                    }
                }
                break;

            case R.id.mul:
                if(op1 == 0){
                    op1 = Integer.parseInt(disp.getText().toString());
                    disp.setText("0");
                    optr = "*";
                }  else {
                    if (continueCalculationl == false){
                        disp.setText("0");
                        optr = "*";
                        continueCalculationl = true;
                    }
                    else {
                        operation();
                        optr = "*";
                        clearDisplay = true;
                    }
                }
                break;

            case R.id.div:
                if(op1 == 0){
                    op1 = Integer.parseInt(disp.getText().toString());
                    disp.setText("0");
                    optr = "/";
                }  else {
                    if (continueCalculationl == false) {
                        disp.setText("0");
                        optr = "/";
                        continueCalculationl = true;
                    }
                    else {
                        operation();
                        optr = "/";
                        clearDisplay = true;
                    }
                }
                break;

            case R.id.equal:
                if(op1 != 0) {
                    continueCalculationl = false;
                    clearDisplay = true;
                    operation();
                    optr = "";
                }
                break;
            default:
                Log.e("Error", "Error - Default case not provided");
        }
    }
}

File: Calc_iOS/Calculator/MainActivity.swift

import UIKit


public class MainActivity : UIViewController {

    
        @IBOutlet weak var outlet_one: UIButton!
        @IBOutlet weak var outlet_two: UIButton!
        @IBOutlet weak var outlet_three: UIButton!
        @IBOutlet weak var outlet_four: UIButton!
        @IBOutlet weak var outlet_five: UIButton!
        @IBOutlet weak var outlet_six: UIButton!
        @IBOutlet weak var outlet_seven: UIButton!
        @IBOutlet weak var outlet_eight: UIButton!
        @IBOutlet weak var outlet_nine: UIButton!
        @IBOutlet weak var outlet_zero: UIButton!
        @IBOutlet weak var outlet_add: UIButton!
        @IBOutlet weak var outlet_sub: UIButton!
        @IBOutlet weak var outlet_mul: UIButton!
        @IBOutlet weak var outlet_div: UIButton!
        @IBOutlet weak var outlet_cancel: UIButton!
        @IBOutlet weak var outlet_equal: UIButton!
        @IBOutlet weak var outlet_display: UILabel!
        @IBOutlet weak var outlet_equation: UILabel!
        @IBOutlet weak var outlet_RC_ID_1000: UIView!
        @IBOutlet weak var outlet_RC_ID_1003: UIView!
        @IBOutlet weak var outlet_RC_ID_1008: UIView!
        @IBOutlet weak var outlet_RC_ID_1013: UIView!
        @IBOutlet weak var outlet_RC_ID_1018: UIView!


    @IBAction func onClick(_ btnPressed: UIView?){
        switch (btnPressed){

        case outlet_zero:
            updateDisp(zero!.title( for: UIControl.State.normal )!)
            break

        case outlet_one:
            updateDisp(one!.title( for: UIControl.State.normal )!)
            break

        case outlet_two:
            updateDisp(two!.title( for: UIControl.State.normal )!)
            break

        case outlet_three:
            updateDisp(three!.title( for: UIControl.State.normal )!)
            break

        case outlet_four:
            updateDisp(four!.title( for: UIControl.State.normal )!)
            break

        case outlet_five:
            updateDisp(five!.title( for: UIControl.State.normal )!)
            break

        case outlet_six:
            updateDisp(six!.title( for: UIControl.State.normal )!)
            break

        case outlet_seven:
            updateDisp(seven!.title( for: UIControl.State.normal )!)
            break

        case outlet_eight:
            updateDisp(eight!.title( for: UIControl.State.normal )!)
            break

        case outlet_nine:
            updateDisp(nine!.title( for: UIControl.State.normal )!)
            break

        case outlet_cancel:
            op1 = 0
            op2 = 0
            continueCalculationl = false
            clearDisplay = true
            disp!.text = "0"
            // RC Note (Java Code) : disp.setHint ( "Perform Operation :)" ) ;    // ToDo - UILabel do not support placeholder property
            equation!.text = ""
            break

        case outlet_add:
            if op1 == 0 {
                op1 = Int(disp!.text!) ?? 0
                disp!.text = "0"
                optr = "+"
            }
            else {
                if continueCalculationl! == false {
                    disp!.text = "0"
                    optr = "+"
                    continueCalculationl = true
                }
                else {
                    operation()
                    optr = "+"
                    clearDisplay = true
                }
            }
            break

        case outlet_sub:
            if op1 == 0 {
                op1 = Int(disp!.text!) ?? 0
                disp!.text = "0"
                optr = "-"
            }
            else {
                if continueCalculationl! == false {
                    disp!.text = "0"
                    optr = "-"
                    continueCalculationl = true
                }
                else {
                    operation()
                    optr = "-"
                    clearDisplay = true
                }
            }
            break


        case outlet_mul:
            if op1 == 0 {
                op1 = Int(disp!.text!) ?? 0
                disp!.text = "0"
                optr = "*"
            }
            else {
                if continueCalculationl! == false {
                    disp!.text = "0"
                    optr = "*"
                    continueCalculationl = true
                }
                else {
                    operation()
                    optr = "*"
                    clearDisplay = true
                }
            }
            break

        case outlet_div:
            if op1 == 0 {
                op1 = Int(disp!.text!) ?? 0
                disp!.text = "0"
                optr = "/"
            }
            else {
                if continueCalculationl! == false {
                    disp!.text = "0"
                    optr = "/"
                    continueCalculationl = true
                }
                else {
                    operation()
                    optr = "/"
                    clearDisplay = true
                }
            }
            break

        case outlet_equal:
            if op1 != 0 {
                continueCalculationl = false
                clearDisplay = true
                operation()
                optr = ""
            }
            break

        default :
            print("Error : Error - Default case not provided")
        }
    }

    
    // MARK: - Main code start
    
    var one : UIButton? = UIButton()
    var two : UIButton? = UIButton()
    var three : UIButton? = UIButton()
    var four : UIButton? = UIButton()
    var five : UIButton? = UIButton()
    var six : UIButton? = UIButton()
    var seven : UIButton? = UIButton()
    var eight : UIButton? = UIButton()
    var nine : UIButton? = UIButton()
    var zero : UIButton? = UIButton()
    var add : UIButton? = UIButton()
    var sub : UIButton? = UIButton()
    var mul : UIButton? = UIButton()
    var div : UIButton? = UIButton()
    var cancel : UIButton? = UIButton()
    var equal : UIButton? = UIButton()
    var disp : UILabel? = UILabel()
    var equation : UILabel? = UILabel()
    var op1 : Int = 0
    var op2 : Int = 0
    var continueCalculationl : Bool? = false
    var clearDisplay : Bool? = false
    var optr : String? = ""

    //  Note - source android method is onCreate()
    override public func viewDidLoad(){
        super.viewDidLoad()

        continueCalculationl = false
        clearDisplay = false
        optr = ""
        op1 = 0
        op2 = 0

        one = outlet_one
        two = outlet_two
        three = outlet_three
        four = outlet_four
        five = outlet_five
        six = outlet_six
        seven = outlet_seven
        eight = outlet_eight
        nine = outlet_nine
        zero = outlet_zero
        add = outlet_add
        sub = outlet_sub
        mul = outlet_mul
        div = outlet_div
        cancel = outlet_cancel
        equal = outlet_equal
        disp = outlet_display
        equation = outlet_equation

    }


    open func operation() -> Void {
        if !( optr  ==  "" ) {
            op2 = Int(disp!.text!) ?? 0
            equation!.text = (equation!.text! + op1.description + " " + optr! + " " + op2.description + " | ")

            if optr  ==  "+" {
                op1 = op1 + op2
                disp!.text = op1.description
            }
            else if optr  ==  "-" {
                op1 = op1 - op2
                disp!.text = op1.description
            }
            else if optr  ==  "*" {
                op1 = op1 * op2
                disp!.text = op1.description
            }
            else if optr  ==  "/" {
                op1 = op1 / op2
                disp!.text = op1.description
            }
            op2 = 0
            continueCalculationl = true
        }
    }

    open func updateDisp(_ str: String?) -> Void {
        //Editable strToUpdate = (Editable)disp.getText();
        //Editable strToUpdate;

        var strToUpdate : String? = ""

        strToUpdate = disp!.text!

        if strToUpdate  ==  "0" {
            strToUpdate = ""
        }

        if clearDisplay! == true {
            clearDisplay = false
            strToUpdate = ""
        }

        strToUpdate = strToUpdate! + str!
        disp!.text = strToUpdate!
    }

}
